habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -597,6 +608,7 @@ def collision_check(
         """
         # Get the offset positions
         num_check_cylinder = len(self._navmesh_offset)
+        # TODO: height 0 is not a good assumption in general. This must be changed to query current navmesh height.
```

Comment:
This would be a breaking change in behavior for the action. So far everything in this PR is transparent. I didn't want to potentially change downstream behaviors.